UI/window-basic-preview.cpp

```diff
@@ -2007,3 +2007,292 @@ OBSBasicPreview *OBSBasicPreview::Get()
+
+	gs_matrix_pop();
+
+	gs_load_vertexbuffer(nullptr);
+
+	gs_technique_end_pass(tech);
+	gs_technique_end(tech);
+}
+
+static void RenderSpacingHelper(int sourceIndex, vec3 &start, vec3 &end,
+				vec3 &viewport)
+{
+	bool horizontal = (sourceIndex == 2 || sourceIndex == 3);
+
+	// If outside of preview, don't render
+	if (!((horizontal && (end.x >= start.x)) ||
+	      (!horizontal && (end.y >= start.y))))
+		return;
+
+	float length = vec3_dist(&start, &end);
+
+	obs_video_info ovi;
+	obs_get_video_info(&ovi);
+
+	float px;
+
+	if (horizontal) {
+		px = length * ovi.base_width;
+	} else {
+		px = length * ovi.base_height;
+	}
+
+	if (px < 1.0f)
+		return;
+
+	OBSBasicPreview *prev = OBSBasicPreview::Get();
+	obs_source_t *source = prev->spacerLabel[sourceIndex];
+	vec3 labelSize, labelPos;
+	vec3_set(&labelSize, obs_source_get_width(source),
+		 obs_source_get_height(source), 1.0f);
+
+	vec3_div(&labelSize, &labelSize, &viewport);
+
+	vec3 labelMargin;
+	vec3_set(&labelMargin, SPACER_LABEL_MARGIN, SPACER_LABEL_MARGIN, 1.0f);
+	vec3_div(&labelMargin, &labelMargin, &viewport);
+
+	vec3_set(&labelPos, end.x, end.y, end.z);
+	if (horizontal) {
+		labelPos.x -= (end.x - start.x) / 2;
+		labelPos.x -= labelSize.x / 2;
+		labelPos.y -= labelMargin.y + (labelSize.y / 2) +
+			      (HANDLE_RADIUS / viewport.y);
+	} else {
+		labelPos.y -= (end.y - start.y) / 2;
+		labelPos.y -= labelSize.y / 2;
+		labelPos.x += labelMargin.x;
+	}
+
+	DrawSpacingLine(start, end, viewport);
+	SetLabelText(sourceIndex, (int)px);
+	DrawLabel(source, labelPos, viewport);
+}
+
+void OBSBasicPreview::DrawSpacingHelpers()
+{
+	if (locked)
+		return;
+
+	OBSBasic *main = OBSBasic::Get();
+
+	if (main->ui->sources->selectionModel()->selectedIndexes().count() > 1)
+		return;
+
+	OBSSceneItem item = main->GetCurrentSceneItem();
+	if (!item)
+		return;
+
+	if (obs_sceneitem_locked(item))
+		return;
+
+	vec2 itemSize = GetItemSize(item);
+	if (itemSize.x == 0.0f || itemSize.y == 0.0f)
+		return;
+
+	matrix4 boxTransform;
+
+	obs_sceneitem_get_box_transform(item, &boxTransform);
+
+	obs_transform_info oti;
+	obs_sceneitem_get_info(item, &oti);
+
+	obs_video_info ovi;
+	obs_get_video_info(&ovi);
+
+	vec3 size;
+	vec3_set(&size, ovi.base_width, ovi.base_height, 1.0f);
+
+	// Init box transform side locations
+	vec3 left, right, top, bottom;
+
+	vec3_set(&left, 0.0f, 0.5f, 1.0f);
+	vec3_set(&right, 1.0f, 0.5f, 1.0f);
+	vec3_set(&top, 0.5f, 0.0f, 1.0f);
+	vec3_set(&bottom, 0.5f, 1.0f, 1.0f);
+
+	// Switch top/bottom or right/left if scale is negative
+	if (oti.scale.x < 0.0f) {
+		vec3_set(&left, 1.0f, 0.5f, 1.0f);
+		vec3_set(&right, 0.0f, 0.5f, 1.0f);
+	}
+
+	if (oti.scale.y < 0.0f) {
+		vec3_set(&top, 0.5f, 1.0f, 1.0f);
+		vec3_set(&bottom, 0.5f, 0.0f, 1.0f);
+	}
+
+	// Decide which side to use with box transform, based on rotation
+	// Seems hacky, probably a better way to do it
+	float rot = oti.rot;
+
+	if (rot >= 90.0f) {
```

Comment:
Can we adjust this 'breakpoint' to 45 instead of 90? Should avoid the near parallel line effect at rotations near multiples of 90

Reply:
I don't think this can be avoided, as far as I know, because the parallel lines happen regardless of what angle you put this at.

Reply:
Here's what I mean

![image](https://user-images.githubusercontent.com/1554753/173735659-dd197dc5-35e0-4a80-be6c-cb555820d0f7.png)

https://user-images.githubusercontent.com/1554753/173735628-ad6c039f-7061-47f0-b746-aa0919d5a272.mp4

Reply:
I see what you mean now.

Reply:
Yep, it works much better now. Thanks for the change suggestion.